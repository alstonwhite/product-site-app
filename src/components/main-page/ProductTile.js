import React from 'react';
import { Link } from "react-router-dom";

import logo from '../../assets/particle-logo2.jpeg';

const ProductTile = ({product}) => {
    return (
      <div className="product-tile" id="product-tile">
        <Link to={`/product/${product.id}`} className = "product-tile__content">  
          <img alt='logo' className="product-tile__img" src={product.image1 ? product.image1.fields.file.url : logo}/>
          <div className="product-tile__name">{product.title}</div>
          <div className="product-tile__price">
              <span>$</span>
              {product.price}
          </div>
        </Link>
      </div>
    );
  };
  
  export default ProductTile;