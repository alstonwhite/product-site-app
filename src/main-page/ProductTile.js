import React from 'react';
import {
  BrowserRouter as Router,
  Switch,
  Route,
  Link
} from "react-router-dom";

import logo from '../logo.svg';

const ProductTile = ({product}) => {
    
    return (
      <div className="product-tile" id="product-tile">
        <Link to={`/product/${product.id}`}>  
          <img alt='logo' className='tile-img' src={logo}/>
          <div className="product-name">{product.productTitle}</div>
          <div className="product-price">
              <span>$</span>
              {product.price}
          </div>
        </Link>
      </div>
    );
  };
  
  export default ProductTile;

  // 