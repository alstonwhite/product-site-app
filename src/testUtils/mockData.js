export const testSections = 
    [
        {
            id: "0",
            slug: "section1",
            sectionTitle: "Section 1",
            active: true
        },
        {
            id: "1",
            slug: "section2",
            sectionTitle: "Section 2",
            active: true
        }
    ]

export const testProducts = 
    [
        {
            id: "0",
            image1: {fields: {file: {url: "1"}}},
            image2: {fields: {file: {url: "2"}}},
            image3: {fields: {file: {url: "3"}}},
            price: 5,
            title: "Product 1",
            description: "Description",
            productSection: {fields: {slug: "section1"}},
            active: true
        },
        {
            id: "1",
            image1: {fields: {file: {url: "1"}}},
            image2: {fields: {file: {url: "2"}}},
            image3: {fields: {file: {url: "3"}}},
            price: 5,
            title: "Product 2",
            description: "Description",
            productSection: {fields: {slug: "section1"}},
            active: true
        },
        {
            id: "2",
            image1: {fields: {file: {url: "1"}}},
            image2: {fields: {file: {url: "2"}}},
            image3: {fields: {file: {url: "3"}}},
            price: 5,
            title: "Product 3",
            description: "Description",
            productSection: {fields: {slug: "section2"}},
            active: true
        }
    ]

export const testProduct = 
    {
        id: 0,
        image1: {fields: {file: {url: "1"}}},
        image2: {fields: {file: {url: "2"}}},
        image3: {fields: {file: {url: "3"}}},
        price: 5,
        title: "Product 1",
        description: "Description",
        productSection: {fields: {slug: "section1"}},
        active: true
    }

export const testCartProduct = 
[{
    id: 0,
    image1: {fields: {file: {url: "1"}}},
    image2: {fields: {file: {url: "2"}}},
    image3: {fields: {file: {url: "3"}}},
    price: 5,
    title: "Product 1",
    description: "Description",
    productSection: {fields: {slug: "section1"}},
    active: true,
    quantity: 1
}]
