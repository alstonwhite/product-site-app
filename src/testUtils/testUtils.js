import React from "react";
import { Router, Route } from "react-router-dom";
import { render } from "@testing-library/react";
import { createMemoryHistory } from "history";

export const renderWithRouter = (ui, 
    {
        path="/",
        route="/",
        history = createMemoryHistory({ initialEntries: [route] }) 
    }={}) => ( 
    render( 
        <Router history={history}> 
            <Route path={path} children={ui} /> 
        </Router> 
    ) 
)