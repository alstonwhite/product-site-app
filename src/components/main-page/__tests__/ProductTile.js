import React from 'react'
import { cleanup, screen } from '@testing-library/react'
import ProductTile from '../ProductTile'
import { testProduct } from '../../../testUtils/mockData'
import { renderWithRouter } from '../../../testUtils/testUtils'


afterEach(cleanup)

test('renders proper product', () => {
    renderWithRouter(<ProductTile product={testProduct}/>, {path: "/"})

    expect(screen.getByText(testProduct.title)).toBeInTheDocument()
})
  
test('snapshot', () => {
    const {container} = renderWithRouter(<ProductTile product={testProduct}/>, {path: "/"})

    expect(container.firstChild).toMatchSnapshot()
})

