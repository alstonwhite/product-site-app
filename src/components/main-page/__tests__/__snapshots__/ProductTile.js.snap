// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div
  class="product-tile"
  data-testid="product-tile"
  id="product-tile"
>
  <a
    class="product-tile__content"
    href="/product/0"
  >
    <img
      alt="logo"
      class="product-tile__img"
      src="1"
    />
    <div
      class="product-tile__name"
    >
      Product 1
    </div>
    <div
      class="product-tile__price"
    >
      <span>
        $
      </span>
      5
    </div>
  </a>
</div>
`;
